### PR TITLE
onboard mugen dataset

### PR DESCRIPTION
**Pull request by sophiazhi**
Summary:
Onboard MUGEN dataset. Changes from the original MUGEN repo include:
- remove dependencies on `jukebox` and MUGEN `audio_vqvae` and place into `data/audio_utils.py`
- replace cmd line `argparse` with `MUGENDatasetArgs` and `DataModuleArgs`
- `data.py` --> `mugen_datamodules.py`. Rename `VideoData` to `MUGENDataModule` and add data transforms.
- `mugen_data.py` --> `mugen_dataset.py`
- remove `data/coinrun/assets` folder and replace with README instructions on how to download assets and a user-configurable asset path argument

Test plan:
In colab [notebook](https://colab.research.google.com/drive/1vzGm8TN-91gMlVr_skLbzW2oUWOyzJtE#scrollTo=cHk_CuGPC1sc)
Recall results match those from original MUGEN implementation.

### CODE REVIEW

File: examples/mugen/data/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This folder contains code for interfacing the MUGEN dataset (https://mugen-org.github.io/data).
```
Comment (langong347) [inline examples/mugen/data/README.md:1]: Perhaps add a summary of what MUGEN dataset is here?

**Review by langong347 — COMMENTED** [review]
I suppose you are still working on removing the assets and figure out instructions for people to download them, right?

File: examples/mugen/data/mugen_datamodules.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+import pytorch_lightning as pl
+import torch
+import torch.distributed as dist
+import torch.utils.data as data
+
+from .mugen_dataset import MUGENDataset, MUGENDatasetArgs
+
+
+class MUGENDataModuleArgs(NamedTuple):
```
Comment (RdoubleA) [inline examples/mugen/data/mugen_datamodules.py:17]: why make this class, how about just using these args in `MUGENDataModule` itself? [FLAVA](https://github.com/facebookresearch/multimodal/blob/a1c136853ebee87bb148e2442991f5430cb40fa2/examples/flava/data/datamodules.py#L60) does something similar

File: examples/mugen/data/mugen_datamodules.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    """
+
+    def __init__(
+        self,
+        mugen_dataset_args: MUGENDatasetArgs,
+        data_module_args: MUGENDataModuleArgs = MUGENDataModuleArgs(),
+        text_transform: Optional[Callable] = None,
+        video_transform: Optional[Callable] = None,
+        audio_transform: Optional[Callable] = None,
+        shuffle=True,
```
Comment (RdoubleA) [inline examples/mugen/data/mugen_datamodules.py:49]: nit: we don't run type checking on examples but since you defined the type for every other argument might as well add it for `shuffle` as well

**Review by RdoubleA — APPROVED** [review]

File: examples/mugen/data/audio_utils.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+# LICENSE file in the root directory of this source tree.
+
+import av
+import numpy as np
+
+
+AUDIO_SAMPLE_RATE = 22050
+AUDIO_SAMPLE_LENGTH = AUDIO_SAMPLE_RATE * 3.2  # each video is 3.2 seconds
+
+
```
Comment (langong347) [inline examples/mugen/data/audio_utils.py:14]: Is this from Jukebox? Can we add a url to the reference code for tracking purpose?

**Review by langong347 — APPROVED** [review]
Looks good. Left a comment.

File: examples/mugen/data/audio_utils.py (hunk 4 above)
Comment (sophiazhi) [reply examples/mugen/data/audio_utils.py:14]: it's from MUGEN in their [audio vqvae](https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/audio_vqvae/hparams.py#L7)

Comment (sophiazhi) [reply examples/mugen/data/audio_utils.py:14]: the load_audio function is from jukebox so i'll add that url too
